src/mock-tooling/index.ts

```diff
@@ -20,36 +20,68 @@ function generateToken(): Token {
 }
 
 /**
- * Mocks the access token endpoint (to request a token).
+ * Parses URL and throws, if URL is neither string nor object
  *
  * @param options
- * @returns {Scope}
  *
  * @throws on parse error of options.url
  */
-export function mockAccessTokenEndpoint(options: MockOptions): void {
-
+function parseUrlOrThrow(options: MockOptions) {
   const parsedUrl = url.parse(options.url);
-
   if (typeof parsedUrl !== 'object' ||
-      typeof parsedUrl.path !== 'string') {
+    typeof parsedUrl.path !== 'string') {
     throw new Error(`Error parsing '${options.url}'`);
   }
+  return parsedUrl;
+}
+
+/**
+ * Mocks the access token endpoint (to request a token).
+ *
+ * @param options
+ * @returns {Scope}
+ *
+ * @throws on parse error of options.url
+ */
+export function mockAccessTokenEndpoint(options: MockOptions): void {
+
+  const parsedUrl = parseUrlOrThrow(options);
 
   nock(`${parsedUrl.protocol}//${parsedUrl.host}`)
-  .post(parsedUrl.path)
+  .post(parsedUrl.path as string) // checked by parseUrlOrThrow
   .times(options.times || Number.MAX_SAFE_INTEGER)
   .query(true)
   .reply(() => {
 
-    // TODO: in the future we want to extrat scopes from body
+    // TODO: in the future we want to extract scopes from body
     const newToken = generateToken();
     tokens.push(newToken);
 
     return [HttpStatus.OK, newToken];
   });
 }
 
+export function mockAccessTokenEndpointWithErrorResponse(options: MockOptions, httpStatus: number, responseBody?: object): void {
```

Comment:
Since the mock does not set a content type we should probably return text instead object. Also the mock would be able to return plain strings.